### PR TITLE
Switch to Dynomodb backed queue

### PR DESCRIPTION
**Pull request by iHiD**
cc @erikschierboom for visibility

### CODE REVIEW

File: lib/orchestrator/application.rb [hunk 1]
```diff
@@ -2,38 +2,34 @@ module Orchestrator
...
-
-      # TOOD: Add a 1sec timer to unlock any locked jobs
-      # that are 15s or older
-
-      # TODO: Remove this temporary thing
-      #enqueue_job!(Job.new("test_runner", 1, 'ruby', 'bob', "s3://exercism-iterations/production/iterations/1182520"))
+      @__client__ = Concurrent::MVar.new(
+        Aws::DynamoDB::Client.new(
+          endpoint: Orchestrator.config.dynamodb_endpoint,
+          profile: "exercism_website",
```
Comment (ErikSchierboom) [inline lib/orchestrator/application.rb:8]: The tooling invoker uses the `exercism_tooling_invoker` profile. Should we change that to `exercism_website`, or this to `exercism_tooling_orchestrator`?

File: lib/orchestrator/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+module Orchestrator
+  class Configuration
+    include Singleton
+
+    def dynamodb_endpoint
+      case env
+      when :production
+        Exercism.config.dynamodb_endpoint
+      else
+        "http://localhost:3039"
```
Comment (ErikSchierboom) [inline lib/orchestrator/configuration.rb:10]: To have this work with docker-compose, we shouldmake this configurable? Something like:

```suggestion
dynamodb_host = ENV.fetch("DYNAMODB_HOST", "127.0.0.1")
"http://#{dynamodb_host}:3039"
```
"http://localhost:3039"

See https://github.com/exercism/tooling-invoker/pull/2/files#diff-6aae0d6010a1060371916ffa8eabbf14R39

File: lib/orchestrator.rb [hunk 3]
```diff
@@ -5,17 +5,23 @@
...
+
+  def self.dynamodb_client
+
   end
 end
 
 # Get a new application on this main thread
 # before sinatra or anything else kicks in
-Orchestrator.application
+#Orchestrator.application
```
[the remark this answers is not in the document]
Comment (iHiD) [reply lib/orchestrator.rb:27]: Ha! :) I'm still working the state out. I'll go through all of these bits and tidy them once I've got a pattern that I'm comfortable with. 

But thank you - @kntsoriano is exactly the same - he's always nagging me about this, and I do appreciate it! :)

File: lib/orchestrator/application.rb (hunk 1 above)
Comment (iHiD) [reply lib/orchestrator/application.rb:8]: This is all getting unified everywhere now, hopefully. Will explain it all when i'm done.

**Pull request merged**